File: lib/limelight/builtin/players/password_box.rb
#- Copyright 2008 8th Light, Inc. All Rights Reserved.
#- Limelight and all included source files are distributed under terms of the GNU LGPL.

module Limelight
  module Builtin
    module Players

      # A Builtin Player that adds the look and behavior of a native password box.  It may be applied in the PropBuilder DSL
      # like so:
      #
      #   my_password_box :players => "password_box"
      #
      # Props including this Player may implement any of the key and focus event hooks:
      #
      #   key_pressed, key_typed, key_released, focus_gained, focus_lost
      #
      module PasswordBox
        class << self

          def extended(prop) #:nodoc:
            password_box = Limelight::UI::Model::Inputs::PasswordBoxPanel.new
            prop.panel.add(password_box)
            prop.password_box = password_box.password_box
          end

        end

        attr_accessor :password_box #:nodoc:

      end

    end
  end
end

File: src/limelight/ui/model/inputs/PasswordBox.java
//- Copyright © 2008-2009 8th Light, Inc. All Rights Reserved.
//- Limelight and all included source files are distributed under terms of the GNU LGPL.

package limelight.ui.model.inputs;

import javax.swing.*;
import java.awt.*;

public class PasswordBox extends JPasswordField
{
  private final PasswordBoxPanel panel;

  public PasswordBox(PasswordBoxPanel panel)
  {
    this.panel = panel;
  }

  public void repaint()
  {
    if(panel != null && panel.getRoot() != null)
      panel.getRoot().addDirtyRegion(panel.getAbsoluteBounds());
  }

  public void repaint(long tm, int x, int y, int width, int height)
  {
    repaint(new Rectangle(x, y, width, height));
  }

  public void repaint(Rectangle dirtyBounds)
  {
    if(panel != null && panel.getRoot() != null)
    {
      Point location = panel.getAbsoluteLocation();
      dirtyBounds.translate(location.x, location.y);
      panel.getRoot().addDirtyRegion(dirtyBounds);
    }
  }

  public boolean isShowing()
  {
    return true;
  }
}
